tModBuildTasks/SynchronizeDirectories.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using Microsoft.Build.Framework;
+
+namespace tModLoader.BuildTasks;
+
+/// <summary>
+/// Copies contents from one folder to another, additionally removing entries in the destination that aren't present in the source.
+/// </summary>
+public class SynchronizeDirectories : TaskBase
+{
+	[Required]
+	public string Source { get; set; } = string.Empty;
+
+	[Required]
+	public string Destination { get; set; } = string.Empty;
+
+	protected override void Run()
+	{
+		Source = Path.GetFullPath(Source);
+		Destination = Path.GetFullPath(Destination);
+
+		var source = new DirectoryInfo(Source);
+		var destination = new DirectoryInfo(Destination);
+
+		if (!source.Exists) {
+			Log.LogError($"Source directory '{Source}' doesn't exist!");
+			return;
+		}
+
+		destination.Create();
+
+		if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows)) {
+			WindowsImplementation();
+		}
+		else {
+			UnixImplementation();
+		}
+	}
+
+	private void WindowsImplementation()
+	{
+		// Robocopy is always present.
+		ExecuteAndWait(
+			"robocopy", @$"""{Source}"" ""{Destination}"" /MIR",
```

Comment:
Since we have .NET at our hands, I'd prefer to use the file API and do it ourselves rather than using `robocopy` or `rsync`

Reply:
I've actually written a fully managed implementation for this task prior, but no matter to the presence of metadata & SIMD'ed hashchecks together with light multi-threading - it ended up feeling about 15 times slower than robocopy and rsync, so I didn't even bother including that as an rsync-less fallback. .NET doesn't seem very good when it comes to streams, but check if you see anything terrible here:

[SynchronizeDirectoriesBuildTask.zip](https://github.com/tModLoader/tModLoader/files/13635951/SynchronizeDirectoriesBuildTask.zip)

Reply:
Neither `rsync` nor `robocopy` compare file contents, only sizes and timestamps by default

Reply:
Pushed the above with integrity checks removed, and some other edits. Seems to average just 10ms now, uh oh.